File: articles/mental-accounting-personal-finance/go/mental_accounting_decision.go
package main

import "fmt"

func debtRepaymentGain(debtInterestRate, savingsRate, repaymentAmount float64) float64 {
	return (debtInterestRate - savingsRate) * repaymentAmount
}

func repayDebtFromLabeledSavings(debtInterestRate, savingsRate, repaymentAmount, labelPenalty float64) bool {
	gain := debtRepaymentGain(debtInterestRate, savingsRate, repaymentAmount)
	return gain > labelPenalty
}

func main() {
	fmt.Printf("Repay debt from labeled savings: %v\n", repayDebtFromLabeledSavings(0.22, 0.02, 1000.0, 150.0))
}

File: articles/mental-accounting-personal-finance/rust/mental_accounting_decision.rs
fn debt_repayment_gain(debt_interest_rate: f64, savings_rate: f64, repayment_amount: f64) -> f64 {
    (debt_interest_rate - savings_rate) * repayment_amount
}

fn repay_debt_from_labeled_savings(
    debt_interest_rate: f64,
    savings_rate: f64,
    repayment_amount: f64,
    label_penalty: f64,
) -> bool {
    debt_repayment_gain(debt_interest_rate, savings_rate, repayment_amount) > label_penalty
}

fn main() {
    println!(
        "Repay debt from labeled savings: {}",
        repay_debt_from_labeled_savings(0.22, 0.02, 1000.0, 150.0)
    );
}
